h/views/api/groups.py

```diff
@@ -81,6 +83,34 @@ def remove_member(group, request):
     return HTTPNoContent()
 
 
+@api_config(route_name='api.group_member',
+            request_method='POST',
+            link_name='group.member.add',
+            description='Add the current user to a group.',
+            effective_principals=security.Authenticated)
+def add_member(request):
```

Comment:
Take a peek at the `remove_member` handler in this view and you'll see it takes an extra param—`group`.

Pyramid views can take a `request` _and_ a `context`. In `remove_member` this `context` param is named `group`.

Thanks to Pyramid traversal (see `h.routes` and the `GroupRoot` in the `h.traversal` module if you're curious about details), view handlers for the route `'/api/groups/{pubid}/members/{user}'` will receive an additional parameter—this really _should be_ a `GroupContext` object, but, probably easier for you, it's currently a `Group` model—because it's not a Context as it should be, this param is named `group` in `remove_member` instead of `context` to denote that it's not actually a proper Context object.

ANYWAY, point being, you can remove the dependency on the `groupfinder` and take the `group` param and you'll have the group in your hot little hands without having to search for it yourself.

There's some exploration to be done here because I can't remember what Pyramid does if the `group` lookup fails in this situation. Does it raise a 404 without your intervention?

Assuming Pyramid _doesn't_ return a 404 if group lookup fails, you'll still need to do the check against the group's authority to make sure it matches `auth_client.authority`.

And you do still need to look up the user as you are doing.